modules/lwjgl/core/src/main/java/org/lwjgl/system/ThreadLocalUtil.java

```diff
@@ -184,21 +189,17 @@ public static void setCapabilities(long capabilities) {
 
     // Ensures FUNCTION_MISSING_ABORT will be called even if no context is current,
     public static void setFunctionMissingAddresses(int functionCount) {
-        // OpenJDK: NULL
-        // GraalVM Native Image: pointer to UnimplementedWithJNIEnvArgument function (see #875)
-        long RESERVED0_NULL = memGetAddress(JNI_NATIVE_INTERFACE);
-
         long ptr = JNI_NATIVE_INTERFACE + CAPABILITIES_OFFSET;
 
         long currentTable = memGetAddress(ptr);
         if (functionCount == 0) {
-            if (currentTable != RESERVED0_NULL) {
+            if (currentTable != RESERVED3_NULL) {
                 FUNCTION_MISSING_ABORT_TABLE = NULL;
                 getAllocator().free(currentTable);
                 memPutAddress(ptr, NULL);
             }
         } else {
-            if (currentTable != RESERVED0_NULL) {
+            if (currentTable != RESERVED3_NULL) {
```

Comment:
What is this check really trying to achieve? If it's about checking whether `reserved3` looks "available" then maybe a better heuristic is to assume that
* `NULL` means available (the common case)
* same value as `reserved0` likely means "available" for something  like GraalVM Native Image which fills `reserved0-3` with some non-null "unimplemented" value.